src/devices/Ina219/Ina219.cs

```diff
@@ -0,0 +1,375 @@
+        /// <remarks>
+        /// This binding does not dispose the passed in I2cDevice.
+        /// </remarks>
+        /// <param name="i2cDevice">The I2cDevice initialized to communicate with the Ina219.</param>
+        public Ina219(I2cDevice i2cDevice)
+        {
+            if (i2cDevice == null)
+            {
+                throw new System.ArgumentNullException(nameof(i2cDevice));
+            }
+
+            PowerLsb = CurrentLsb = 1F;
+
+            _i2cDevice = i2cDevice;
+        }
+
+        /// <summary>
+        /// Construct an Ina219 device using an I2cConnectionSettings.
+        /// </summary>
+        /// <remarks>
+        /// This binding creates an I2cDevice ufor communication with the Ina219. The I2cDevice is disposed when then Ina219 is disposed.
+        /// </remarks>
+        /// <param name="settings">The I2cConnectionSettings object initialized with the appropiate settings to communicate with the Ina219.</param>
+        public Ina219(I2cConnectionSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new System.ArgumentNullException(nameof(settings));
+            }
+
+            PowerLsb = CurrentLsb = 1F;
+            _calibrationValue = 0;
+
+            _disposeI2cDevice = true;
+            _i2cDevice = I2cDevice.Create(settings);
+        }
+
+        /// <summary>
+        /// Reset the Ina219 to default values;
+        /// </summary>
+        public void Reset()
+        {
+            ushort regValue;
+
+            // Reset the device by sending a value to the configuration register with the reset but set.
+            WriteRegister(Register.Configuration, (ushort)ConfigurationFlags.Rst);
+
+            // Read back the values from the configuration register which should now be default.
+            regValue = ReadRegister(Register.Configuration, 1000);
+
+            PowerLsb = PowerLsb = 1F;
+            _calibrationValue = 0;
+
+            // Get the current values for the Bus and Shunt ADC settings.
+            _busAdcResSamp = (AdcResolutionOrSamples)((regValue & (ushort)ConfigurationFlags.BadcMask) >> 4);
+            _shuntAdcResSamp = (AdcResolutionOrSamples)(regValue & (ushort)ConfigurationFlags.SadcMask);
+        }
+
+        /// <summary>
+        /// Set the Ina219 operating mode.
+        /// </summary>
+        /// <param name="mode">The operating mode to switch to.</param>
+        public void SetOperatingMode(OperatingMode mode)
+        {
+            ushort regValue = ReadRegister(Register.Configuration);
+
+            regValue &= (ushort)ConfigurationFlags.ModeMask ^ 0xFFFF;
+            regValue |= (ushort)mode;
+
+            WriteRegister(Register.Configuration, regValue);
+        }
+
+        /// <summary>
+        /// Set the Ina219 Bus voltage range.
+        /// </summary>
+        /// <param name="range">The voltage range to switch to.</param>
+        public void SetBusVoltageRange(BusVoltageRange range)
```

Comment:
I'd make this a property (or make this private and call it from property)

Reply:
Doesn’t this violate the property vs function guideline. I.e. 
Takes some time and so is more than a field access and you may want an asynchronous version in a future version.
Can throw an exception etc.

Reply:
If the time is just I2C communication this is fine. If there is some extra delays or some heavier work then this should stay as function